HealthyPlanners/Urban-Sherpa
# Login route

- handled post request to /login
- Token with a 1-hour expiration is generated. I didn't change the 1 hour timing because I read that JWT expiry should commonly be no longer than 15 minutes: [Is refreshing an expired JWT token a good strategy?](https://security.stackexchange.com/questions/119371/is-refreshing-an-expired-jwt-token-a-good-strategy#:~:text=After%20authenticating%2C%20hand%20out%20a,out%20a%20new%20JWT%20token.). I did not implement this either, because the jsonwebtoken creators write **"First of all, we recommend you to think carefully if auto-refreshing a JWT will not introduce any vulnerability in your system. We are not comfortable including this as part of the library"**. So it appears this is a subject to discuss.

To test:
1. in Postman, send a post request to /api/user/login with an email and password that exists in the db. ex: **email: alex2@alex.com password: test123**.
2. when post request passes, response should include a cookie with a new JWT, and the email from the request.
If email or password is incorrect, the response should say 'Incorrect Email or Password' in both cases.
